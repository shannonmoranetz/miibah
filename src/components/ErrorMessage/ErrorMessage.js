import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const ErrorMessage = (props) => {
  return (
    <div className="ErrorMessage">
      <h2>error</h2>
      <p>
        uh oh, encountered an error... please try again!
      </p>
      <Link to='/'>go back</Link>
    </div>
  );
}

export default ErrorMessage;

ErrorMessage.propTypes = {
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object
}