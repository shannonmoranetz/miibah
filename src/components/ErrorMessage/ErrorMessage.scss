.ErrorMessage {
  font-family: $heading-font;
  color: white;
  text-align: center;
  font-size: 30px;
}