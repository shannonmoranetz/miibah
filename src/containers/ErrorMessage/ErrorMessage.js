import React, {Component} from 'react';
import { connect } from 'react-redux';

class ErrorMessage extends Component {
  render() {
    return (
      <div className="ErrorMessage">
        <h2>error</h2>
        <p>
          the search returned no results. 
          please try again with a different query.
        </p>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  error: state.error
});

export default (connect(mapStateToProps)(ErrorMessage));