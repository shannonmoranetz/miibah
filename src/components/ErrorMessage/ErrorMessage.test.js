import React from 'react';
import { ErrorMessage } from './ErrorMessage';
import { shallow } from 'enzyme';

describe('ErrorMessage', () => {
  let wrapper; 
  beforeEach(() => {
    wrapper = shallow(<ErrorMessage />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});