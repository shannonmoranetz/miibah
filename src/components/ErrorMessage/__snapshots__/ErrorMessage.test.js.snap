// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should match the snapshot 1`] = `
<div
  className="ErrorMessage"
>
  <h2>
    error
  </h2>
  <p>
    sorry! there was an error. please try again.
  </p>
  <Link
    replace={false}
    to="/"
  >
    go back
  </Link>
</div>
`;
